modules/land.js

```diff
@@ -8,11 +9,65 @@ function Land()
 }
 
 Land.prototype = {
-  draw: function(ctx)
+  draw: function()
   {
-    ctx.globalAlpha = 0.8;
-    ctx.fillStyle="#ffffff";
-    ctx.fillRect(0, 0, this.width, this.height);
-    ctx.globalAlpha = 1;
+    // remove dead organisms
+    this.removeDeadOrganisms();
+
+    this.ctx.globalAlpha = 0.8;
+    this.ctx.fillStyle="#ffffff";
+    this.ctx.fillRect(0, 0, this.width, this.height);
+    this.ctx.globalAlpha = 1;
+
+    this.drawOrganisms();
+    this.drawFoods();
+  },
+
+  removeDeadOrganisms: function()
+  {
+    for (var i = this.population.length-1; i >= 0; i--) {
+      if (!this.population[i].alive) {
+        this.population.splice(i, 1);
+      }
+    }
+  },
+
+  drawOrganisms: function()
+  {
+    for (var i in this.population)
+    {
+      var organism = this.population[i];
+
+      if (!debug.isPaused())
+      {
+        organism.move(this);
```

Comment:
I'll move it inside organism.draw - same with food.update, I'll move it in food.draw